Mathlib/Dynamics/TopologicalEntropy/DynamicalUniformity.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Order.Interval.Finset.Nat
+import Mathlib.Order.OmegaCompletePartialOrder
+import Mathlib.Topology.UniformSpace.Basic
+
+/-!
+# Dynamical uniformities
+Bowen-Dinaburg's definition of topological entropy of a transformation `T` in a metric space
+`(X, d)` relies on the so-called dynamical balls. These balls are sets
+`B (x, ε, n) = { y | ∀ k < n, d(T^[k] x, T^[k] y) < ε }`.
+
+We implement Bowen-Dinaburg's definitions in the more general context of uniform spaces. Dynamical
+balls are replaced by what we call dynamical uniformities. This file collects all general lemmas
+about these objects.
+
+## Main definitions
+- `DynamicalUni`: dynamical uniformity associated with a given transformation `T`, uniformity `U`
+and time `n`.
```

Comment:
The uniformity is the uniform structure, but here it's not what you are talking about, you are creating a neighborhood of the diagonal, called an entourage. So the name should probably be `dynEntourage`. Note that it should be lower case per our naming convention.

Reply:
Done.

I tried to follow Topology.UniformSpace.Basic, which weirdly enough, manages to never use the word "entourage" outside of the docstrings...